### PR TITLE
Add new condition to DL2365Groupupgrade script

### PR DESCRIPTION
**Pull request by hazemembaby**
**Issue:**
Describe what the issue is you are addressing

- I've added new condition to test DL owners with RecipientTypeDetails other than UserMailbox, MailUser
- I've modified the EAP condition to work case sensitively while comparing DL PrimarySmtpAddress& EnabledPrimarySMTPAddressTemplate
- Modified readme file by change title O365 to Microsoft 365

**Reason:**
Describe what the reason is for making the change

To address these new conditions I've mentioned earlier that might fail DL to M365 group upgrade

**Fix:**
Short description of the fix

- I've added new condition to test DL owners with RecipientTypeDetails other than UserMailbox, MailUser
- I've modified the EAP condition to work case sensitively while comparing DL PrimarySmtpAddress& EnabledPrimarySMTPAddressTemplate
- Modified readme file by change title O365 to Microsoft 365

**Validation:**
Provide if applicable
image

### CODE REVIEW

File: M365/src/DLT365Groupsupgrade.ps1 [hunk 1]
```diff
@@ -121,9 +121,9 @@ Function Debuggroupnesting {
     $ParentDGroups = @()
     try {
         $alldgs = Get-DistributionGroup -ResultSize unlimited -ErrorAction Stop
-        $CurrentDescription = "Retrieving All DGs in the EXO directory"
-        $CurrentStatus = "Success"
-        log -Function "Retrieve All DGs" -CurrentDescription $CurrentDescription -CurrentStatus $CurrentStatus
+        #$CurrentDescription = "Retrieving All DGs in the EXO directory"
```
Comment (bill-long) [inline M365/src/DLT365Groupsupgrade.ps1:124]: Did you intend to leave these lines commented out? Can we remove them if they are not needed?

File: M365/src/DLT365Groupsupgrade.ps1 [hunk 2]
```diff
@@ -132,9 +132,9 @@ Function Debuggroupnesting {
     foreach ($parentdg in $alldgs) {
         try {
             $Pmembers = Get-DistributionGroupMember $($parentdg.Guid.ToString()) -ErrorAction Stop
-            $CurrentDescription = "Retrieving: $parentdg members"
-            $CurrentStatus = "Success"
-            log -Function "Retrieve Distribution Group membership" -CurrentDescription $CurrentDescription -CurrentStatus $CurrentStatus
+            #$CurrentDescription = "Retrieving: $parentdg members"
```
Comment (bill-long) [inline M365/src/DLT365Groupsupgrade.ps1:135]: More commented out code.

**Review by bill-long — CHANGES_REQUESTED** [review]

File: M365/src/DLT365Groupsupgrade.ps1 (hunk 1 above)
Comment (hazemembaby) [reply M365/src/DLT365Groupsupgrade.ps1]: No I didn't intend to comment these lines, I uncommented them and pushed the change

File: M365/src/DLT365Groupsupgrade.ps1 (hunk 2 above)
Comment (hazemembaby) [reply M365/src/DLT365Groupsupgrade.ps1]: I did intend to remove these by commenting them and I have removed them totally now

**Review by bill-long — APPROVED** [review]

**Pull request merged**